docs/nodejs.md

```diff
@@ -0,0 +1,250 @@
+Console. It can be viewed in Dev Mode or the Trace Explorer Details.
+- A **Captured Warning** is one instance of a string in Node.js that is sent to
+Serverless Console, much like a Captured Error.
+- A **Tag** is a key/value-pair that can be set on the Trace or an individual
+Captured Error or Captured Warning, and sent to Serverless Console. Tags can be
+viewed on the Trace Explorer Details and Dev Mode.
+
+## Compatibility
+
+While Serverless Console is developed by the makers of the Serverless Framework,
+the entire Serverless Console product and this SDK are 100% agnostic of the
+deployment tool you use. Serverless Console and this SDK work just as well with
+Terraform, CDK, SAM, Pulumi, etc, as as they do with Serverless Framework.
+
+## Installation
+
+**Install the package**
+
+When Tracing is enabled in Servelress Console, an  AWS Lambda Layer is added to
+your AWS Lambda function with the `@serverless/sdk` package. If you are running
+the handler with Tracing enabled on AWS Lambda only, then you can skip this
+step.
+
+If you will run this on AWS Lambda without Serverless Console Tracing enabled,
+or running it on a different runtime, like locally, then you'll need to add the
+`@serverless/sdk` package.
+
+```
+npm install @serverless/sdk --save
+# or
+yarn add @serverless/sdk
+```
+
+**Using a bundler**
+
+If you use a bundler, like esbuild, the AWS Lambda Layer for Serverless Console
+will not be able to auto-instrument traces and spans on your handler. To enable
+auto-instrumentation of spans and traces, you will need to manually add the
+AWS-specific auto-instrumentation library and initiate auto-instrumentation.
+
+Install the `@serverless/aws-lambda-sdk` package locally. This replaces
+the need for the `@serverless/sdk` package, so you do not need both.
+
+```
+npm install @serverless/aws-lambda-sdk --save
+# or
+yarn add @serverless/aws-lambda-sdk
+```
+
+Use the following methods to instrument the AWS client libraries and Express.js.
+
+```javascript
+const serverlessSdk = require("@serverless/aws-lambda-sdk");
+
+// Instrument AWS SDK v2
+serverlessSdk.instrumentation.awsSdkV2.install(AWS)
+
+// Instrument AWS SDK v3 client
+serverlessSdk.instrumentation.awsSdkV3Client.install(client)
+
+// instruments Express.js
+serverlessSdk.instrumentation.expressApp.install(expressApp)
+```
+
+**Enable Tracing, Logging, and Dev Mode**
+
+The SDK will merely generate the necessary Tags, Captured Errors, and Captured
+Warnings; however, Tracing, Logging, and Dev Mode must be enabled on your org on
+Serverless Console in order for the data to be ingested.
+
+[Enable Tracing, Logging, and Dev Mode](/console/docs/integrations/enable-monitoring-features)
+
+## Usage
+
+The package does not require any configuration as the credentials are
+automatically set on the AWS Lambda function environment variables when Tracing
+is enabled in Serverless Console.
+
+To use the Serverless SDK you must require the `@serverless/sdk` method in your
```

Comment:
We may add note, that this package doesn't have to be installed and bundled for Lambda, that the extension will ensure it in the Lambda environment (but also bundling won't break it)

Reply:
In the **Install the package&** section I added a note about how bundling the package is optional. I added a note in there that bundling won't break it.

Reply:
I've just realized that not bundling the SDK is dangerous due to how we instrument lambda.

It's possible to have integrated lambda published without our extension layer (e.g. if some does redeployment manually and alter the layer section it will remove our extension), then if code attempts to require our SDK it'll crash, and we have no means to mitigate that.

Not sure though what's the best way to communicate that, and how likely such situations are